src/type_checker.cpp

```diff
@@ -61,22 +87,28 @@ void TypeChecker::Visit(ProgramNode& program) {
 
 void TypeChecker::Visit(IfStmtNode& if_stmt) {
   if_stmt.predicate->Accept(*this);
+  body_types.push_back(BodyType::kLoop);
```

Comment:
Why is `If` statement a `kLoop`, but not a `kNone`?

Reply:
Oops. :cry: You're right. The `If` statement should not have a body type. I'll address it.